src/cpp/src/llm_pipeline_static.cpp

```diff
@@ -16,6 +17,24 @@
 
 namespace {
 
+std::shared_ptr<ov::Model> cvt_kvcache_to_fp16(const std::shared_ptr<ov::Model>& model) {
+    ov::preprocess::PrePostProcessor ppp(model);
+
+    for (auto tensor : model->inputs()) {
+        if (tensor.get_any_name().find("past_key") != std::string::npos) {
+            ppp.input(tensor.get_any_name()).tensor().set_element_type(ov::element::Type_t::f16);
+        }
+    }
+
+    for (auto tensor : model->outputs()) {
+        if (tensor.get_any_name().find("present") != std::string::npos) {
+            ppp.output(tensor.get_any_name()).tensor().set_element_type(ov::element::Type_t::f16);
```

Comment:
Why do the cache need to be converted to `f16` instead of preserving the original precision?

Reply:
As cache should be copied from device to host on every iteration, making it f16 (having less byte) makes it faster on NPU

Reply:
I suspect a quantized model would have quantized cache which has fewer bytes compared to f16.

Reply:
KV cache tensors cannot be compressed via static quantization, so that even for compressed model they will be in float precision